superset-frontend/src/pages/DvtHome/index.tsx

```diff
@@ -16,32 +16,172 @@
+  formatFunction: FormatFunction,
+  setDataFunction: React.Dispatch<React.SetStateAction<CardDataProps[]>>,
+) => {
+  try {
+    const response = await fetch(url);
+    const data = await response.json();
+    setDataFunction(formatFunction(data).sort((a, b) => a.id - b.id));
+  } catch (error) {
+    console.error('API request failed:', error);
+  }
+};
+
+const formatDashboardData: FormatFunction = data =>
+  data.result.slice(0, 5).map(item => ({
+    id: item.id,
+    title: item.dashboard_title,
+    label: item.changed_by_name,
+    description: `Modified ${item.changed_on_delta_humanized}`,
+    isFavorite: item.published,
+    link: item.url,
+  }));
+
+const formatChartData: FormatFunction = data =>
+  data.result.slice(0, 5).map(item => ({
+    id: item.id,
+    title: item.slice_name,
+    label: item.changed_by_name,
+    description: `Modified ${item.created_on_delta_humanized}`,
+    isFavorite: item.is_managed_externally,
+    link: item.url,
+  }));
+
+const formatSavedQueriesData: FormatFunction = data =>
+  data.result.slice(0, 5).map(item => ({
+    id: item.id,
+    title: item.label,
+    label: item.description,
+    description: `Ran ${item.last_run_delta_humanized}`,
+    isFavorite: null,
+    link: '',
+  }));
+
+const formatRecentData: FormatFunction = data =>
+  data.result.slice(0, 5).map(item => ({
+    id: Math.floor(item.time),
+    title: item.item_title,
+    label: '',
+    description: `Modified ${item.time_delta_humanized}`,
+    isFavorite: null,
+    link: item.item_url,
+  }));
+
 function DvtWelcome() {
   const [openCalendar, setOpenCalendar] = useState<boolean>(true);
   const [calendar, setCalendar] = useState<Moment | null>(null);
+  const [recentData, setRecentData] = useState<CardDataProps[]>([]);
+  const [dashboardData, setDashboardData] = useState<any[]>([]);
+  const [chartData, setChartData] = useState<any[]>([]);
+  const [savedQueriesData, setSavedQueriesData] = useState<CardDataProps[]>([]);
+
+  const handleSetFavorites = (
+    id: number,
+    isFavorite: boolean,
+    title: string,
+  ) => {
+    fetch(
+      `/superset/favstar/${title}/${id}/${isFavorite ? 'unselect' : 'select'}/`,
+    ).then(res => {
+      if (res.status === 200) {
+        if (title === 'Dashboard') {
+          const findItem = dashboardData.find(item => item.id === id);
+          const withoutItemData = dashboardData.filter(item => item.id !== id);
+
+          setDashboardData(
+            [...withoutItemData, { ...findItem, isFavorite: !isFavorite }].sort(
+              (a, b) => a.id - b.id,
+            ),
+          );
+        }
+        if (title === 'slice') {
+          const findItem = chartData.find(item => item.id === id);
+          const withoutItemData = chartData.filter(item => item.id !== id);
+
+          setChartData(
+            [...withoutItemData, { ...findItem, isFavorite: !isFavorite }].sort(
+              (a, b) => a.id - b.id,
+            ),
+          );
+        }
```

Comment:
'Dashboard' ve 'slice' altındaki ikise benziyor, yeni function callback yapman gerekiyor. Minify için